build-tools/packages/build-tools/src/common/npmPackage.ts

```diff
@@ -490,11 +489,28 @@ export function updatePackageJsonFile(
 	packageTransformer: (json: PackageJson) => void,
 ): void {
 	const packagePath = path.join(packageDir, "package.json");
-	const indentation = detectIndent(packagePath).indent || "\t";
-	const pkgJson: PackageJson = readJsonSync(packagePath);
+	const [pkgJson, indent] = readPackageJsonAndIndent(packagePath);
```

Comment:
Incidentally, the old code had a bug. It was detecting the indentation of the _path_, not the JSON itself. 🤣  Thank goodness for defaults I guess? But also bad me for somehow not testing this? \<sigh>

Reply:
I added a couple of test cases. I'm pretty new to testing so curious what you think.